pkg/sqlutil/interceptor.go

```diff
@@ -0,0 +1,101 @@
+package sqlutil
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/jmoiron/sqlx"
+)
+
+var _ Queryer = &InterceptedQueryer{}
+
+type Interceptor func(ctx context.Context, do func(context.Context) error, query string, args ...any) error
```

Comment:
Does the `database/sql` package or `sqlx` package provide an Interceptor type + APIs we can use? Seems like we aren't the only ones who would want this functionality as well.

Reply:
I had the same thought but couldn't find one for sqlx